test/sanity/code-smell/docs-build.requirements.in

```diff
@@ -1,9 +1,9 @@
 jinja2
 pyyaml
 resolvelib < 0.10.0
-sphinx == 4.2.0
+sphinx == 5.3.0
 sphinx-notfound-page
 sphinx-ansible-theme
 straight.plugin
-rstcheck < 4  # match version used in other sanity tests
+rstcheck < 6  # rstcheck 6.x has problem with rstcheck.core triggered by include files w/ sphinx directives https://github.com/rstcheck/rstcheck-core/issues/3
```

Comment:
We should probably update the one used by the changelog sanity test as well:

https://github.com/ansible/ansible/blob/875a4ec2428f289cbeb855ccb4e099e67b863813/test/lib/ansible_test/_data/requirements/sanity.changelog.in#L1

Reply:
Both the changelog sanity test and the package-data sanity test restrict docutils to `< 0.18`, while the docs-build and the rstcheck sanity tests use 0.18.1.

I'm not sure why the restriction for `< 0.18` is there; antsibull-changelog itself does not have this restriction. Maybe it's time to drop it? Both say that it shoul dmatch the one of the docs-build sanity test, but that no longer has a restriction on docutils.

Reply:
oh good catch!

Reply:
@felixfontein Can you think of a reason why we'd want to keep the `changelog` sanity test requirements in sync with the docs related tests?

Reply:
@mattclay one reason would be to minimize download size of the container image: if all tests use the same version of rstcheck and docutils, the compressed container image will be smaller since compression can eliminate identical copies of the same files (assuming the requirements are installed in different places). Also I don't see any reason why the other sanity tests should have *different* requirements on rstcheck and docutils, if they have no explicit reason (like known incompatibilities).

The only restriction on rstcheck I can think of is `< 6` for the same reason for all the tests, namely because of https://github.com/rstcheck/rstcheck-core/issues/3.

Reply:
That rstcheck issue shouldn't affect the changelog sanity test, since it doesn't use sphinx.

Reply:
antsibull-changelog supports the same rstcheck versions as antsibull-docs, so `< 6` is fine. (They also support 6, but 6 has this issue with some valid RST content...)